### PR TITLE
Chore: Intro to Storybook updates for 6.0 (react version)

### PR DESCRIPTION
**Pull request by jonniebigodes**
With this pull request the React version of the Intro to Storybook tutorial is updated for 6.0.

What was done:
- Code snippets and wording is updated to include features introduced with 6.0.
- The change proposed in [here](https://github.com/chromaui/learnstorybook.com/pull/325#issuecomment-649185617) is now addressed. I've clarified a bit and added some context.
- The using addons page was updated for controls. Went with a simple example based of the official documentation, to avoid overcomplicating too much as this is a introductory tutorial.
- Links to the official documentation were updated. 
- Links for the component-driven.org are in place.
- The issue raised in #351 should not present itself.

What is not done:

- I'm experiencing an issue with the create addons bonus chapter. I took the entire day today to try and check what might be wrong and couldn't come up with a solution. The problem presents itself as soon as the user adds the addon to his/hers Storybook `main.js` file and issues `yarn storybook` he/she will be presented with the following message:

```bash
yarn run v1.22.4
$ start-storybook -p 6006 -s public
info @storybook/react v6.0.12
info 
info => Loading static files from: /Users/myuser/Documents/my-folder/storybook-related/learn_storybook_with_6_0/public .
info => Loading presets
ERR! Addon value should end in /register OR it should be a valid preset https://storybook.js.org/docs/presets/introduction/
ERR! ./.storybook/design-addon/register.js
info => Loading presets
ERR! Addon value should end in /register OR it should be a valid preset https://storybook.js.org/docs/presets/introduction/
ERR! ./.storybook/design-addon/register.js
info => Loading config/preview file in "./.storybook".
info => Loading config/preview file in "./.storybook".
info => Adding stories defined in ".storybook/main.js".
info => Loading Webpack configuration from `node_modules/react-scripts`
info => Removing existing JavaScript and TypeScript rules.
info => Modifying Create React App rules.
info => Using default Webpack setup.
webpack built bbac42f44f3b9054aa0a in 9708ms
```
I've tested it in both my windows box and mac and the same result was yelded. 
- The updated assets for 6.0 are not added yet, wanted to get the wording/snippets out first. 
- The learnstorybook-code repo is going to be updated shortly.

The code used for the documentation is housed [here](https://github.com/jonniebigodes/intro-storybook-with-storybook-6-0).

Feel free to provide feedback.

### CODE REVIEW

File: content/intro-to-storybook/react/en/composite-component.md [hunk 1]
```diff
@@ -81,29 +72,58 @@ export const defaultTasksData = [
   { ...taskData, id: '5', title: 'Task 5' },
   { ...taskData, id: '6', title: 'Task 6' },
 ];
-
 export const withPinnedTasksData = [
   ...defaultTasksData.slice(0, 5),
   { id: '6', title: 'Task 6 (pinned)', state: 'TASK_PINNED' },
 ];
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/composite-component.md:78]: I think we should drop these exports and instead just reuse the args.

File: content/intro-to-storybook/react/en/composite-component.md [hunk 2]
```diff
@@ -81,29 +72,58 @@ export const defaultTasksData = [
...
-<a href="https://storybook.js.org/addons/introduction/#1-decorators"><b>Decorators</b></a> are a way to provide arbitrary wrappers to stories. In this case we’re using a decorator `key` on the default export to add some `padding` around the rendered component. They can also be used to wrap stories in “providers” –i.e. library components that set React context.
+<a href="https://storybook.js.org/docs/react/writing-stories/decorators"><b>Decorators</b></a> are a way to provide arbitrary wrappers to stories. In this case we’re using a decorator `key` on the default export to add some `padding` around the rendered component. They can also be used to wrap stories in “providers” –i.e. library components that set React context.
 </div>
 
 `taskData` supplies the shape of a `Task` that we created and exported from the `Task.stories.js` file. Similarly, `actionsData` defines the actions (mocked callbacks) that a `Task` component expects, which the `TaskList` also needs.
 
 Now check Storybook for the new `TaskList` stories.
 
+<div class="aside">
+TODO: video needs to be updated for 6.0
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/composite-component.md:124]: I guess we'll need to address this before merging?

File: content/intro-to-storybook/react/en/simple-component.md [hunk 3]
```diff
@@ -46,15 +46,12 @@ Below we build out Task’s three test states in the story file:
...
 
-export default {
-  component: Task,
-  title: 'Task',
-  // Our exports that end in "Data" are not stories.
-  excludeStories: /.*Data$/,
+export const actionsData = {
+  onPinTask: { action: 'onPinTask' },
+  onArchiveTask: { action: 'onArchiveTask' },
 };
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/simple-component.md:55]: Can we use argtypes for the actions?

I'm not sure what's better, either we could:
 - Define a global `argTypeRegex` in `preview.js` or
 - Add action annotations per argTypes in each component.

WDYT @jonniebigodes ?

File: content/intro-to-storybook/react/en/simple-component.md [hunk 4]
```diff
@@ -90,19 +106,28 @@ To tell Storybook about the component we are documenting, we create a `default`
...
 
-`action()` allows us to create a callback that appears in the **actions** panel of the Storybook UI when clicked. So when we build a pin button, we’ll be able to determine in the test UI if a button click is successful.
+As we have multiple permutations of our component, it's convenient to assign it to a `Template` variable. Introducing this pattern in your stories will reduce the amount of code you need to write and maintain.
 
-As we need to pass the same set of actions to all permutations of our component, it is convenient to bundle them up into a single `actionsData` variable and use React's `{...actionsData}` props expansion to pass them all at once. `<Task {...actionsData}>` is equivalent to `<Task onPinTask={actionsData.onPinTask} onArchiveTask={actionsData.onArchiveTask}>`.
+<div class="aside">
+
+Take in consideration that `Template.bind({})` is a [standard JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) technique for making a copy of a function. We recommend this technique, so each exported story can set its own properties on it.
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/simple-component.md:117]:
```suggestion
`Template.bind({})` is a [standard JavaScript](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/bind) technique for making a copy of a function. We use this technique to allow each exported story to set its own properties, but use the same implementation.
```

File: content/intro-to-storybook/react/en/simple-component.md [hunk 5]
```diff
@@ -90,19 +106,28 @@ To tell Storybook about the component we are documenting, we create a `default`
...
+
+</div>
+
+Arguments or [`args`](https://storybook.js.org/docs/react/writing-stories/args) for short, are extremely helpful for editing our components. They allow us to edit our components without restarting Storybook. Once the [`args`](https://storybook.js.org/docs/react/writing-stories/args) value changes so does the component.
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/simple-component.md:121]:
```suggestion
Arguments or [`args`](https://storybook.js.org/docs/react/writing-stories/args) for short, are extremely helpful for editing our components. They allow us to edit our components with the controls addon without restarting Storybook. Once the [`args`](https://storybook.js.org/docs/react/writing-stories/args) value changes so does the component.
```

File: content/intro-to-storybook/react/en/simple-component.md [hunk 6]
```diff
@@ -117,23 +142,31 @@ Start by changing your Storybook configuration file (`.storybook/main.js`) to th
...
-After completing the change above, inside the `.storybook` folder, add a new file called `preview.js` with the following:
+After completing the change above, inside the `.storybook` folder, change the `preview.js` to the following:
 
 ```javascript
 // .storybook/preview.js
 
 import '../src/index.css';
+
+export const parameters = {
+  actions: { argTypesRegex: '^on[A-Z].*' },
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/simple-component.md:160]: Oh! If we did this we don't need to add the actions args.

File: content/intro-to-storybook/react/en/using-addons.md [hunk 7]
```diff
@@ -14,149 +14,227 @@ everybody in your team. If you've been following along with this tutorial linear
...
+<div class="aside">
+TODO: add video or image of the initial controls setup showing
+</div>
 
 ## Addons Evolve Your Storybook's Scope
 
-Not only does your Storybook instance serve as a wonderful [CDD environment](https://blog.hichroma.com/component-driven-development-ce1109d56c8e), but now we're providing an interactive source of documentation. PropTypes are great, but a designer or somebody completely new to a component's code will be able to figure out its behavior very quickly via Storybook with the knobs addon implemented.
+Not only does your Storybook instance serve as a wonderful [CDD environment](https://www.componentdriven.org/), but now we're providing an interactive source of documentation and usability. PropTypes and args are great, but a designer or somebody completely new to a component's code will need to be able to figure out its behavior very quickly via Storybook.
```
Comment (tmeasday) [inline content/intro-to-storybook/react/en/using-addons.md:166]:
```suggestion
Not only does your Storybook instance serve as a wonderful [component-driven development environment](https://www.componentdriven.org/), but now we're providing an interactive source of documentation and usability. PropTypes and args are great, but a designer or somebody completely new to a component's code will need to be able to figure out its behavior very quickly via Storybook.
```

(Let's not use "CDD" any more)

File: content/intro-to-storybook/react/en/composite-component.md (hunk 1 above)
Comment (tmeasday) [reply content/intro-to-storybook/react/en/composite-component.md]: @jonniebigodes maybe we can figure out the jest thing in Slack? It sounds familiar.

In your test you should be able to use `Default.args.tasks` just fine!

File: content/intro-to-storybook/react/en/composite-component.md (hunk 2 above)
Comment (tmeasday) [reply content/intro-to-storybook/react/en/composite-component.md:126]: 😆

File: content/intro-to-storybook/react/en/simple-component.md (hunk 3 above)
Comment (tmeasday) [reply content/intro-to-storybook/react/en/simple-component.md]: Awesome!

File: content/intro-to-storybook/react/en/simple-component.md (hunk 6 above)
Comment (tmeasday) [reply content/intro-to-storybook/react/en/simple-component.md:145]: Yeah, perhaps we need some text in that spot to explain what's happening, WDYT?

**Comment by ndelangen** [comment]
@tmeasday should I put this on my tasklist / engineering board?

**Comment by tmeasday** [comment]
Chuck it on and let's discuss tomorrow

**Comment by jonniebigodes** [comment]
With the latest commit the videos and the images used in the test and deploy chapters were updated for 6.0.

The simple component was changed to include the feedback about parameters.

Feel free to provide feedback

File: content/intro-to-storybook/react/en/composite-component.md (hunk 1 above)
Comment (jonniebigodes) [reply content/intro-to-storybook/react/en/composite-component.md]: @tmeasday i've updated the test in the composite chapter. The initial issue that i had with jest and fsevents is now fixed. After a bit of research it seems that deleting both `yarn.lock` and `node_modules` and then issue `yarn` to install the dependencies works in some cases, it did for me, but can't promise that this is a silver bullet (pardon the bad pun), probably something might be wrong with the cra templated that is installed. 

One item that needs a bit of attention, it seems that the actions are not propagated, running the tests now yield a warning mentioning undefined but they are marked as required. Doing a simple test i was able to bypass it through removing the required from the proptype. It didn't have impact on the snapshots and the actual tests that we're mentioning. 

Let me know if you're ok by changing this?

File: content/intro-to-storybook/react/en/simple-component.md (hunk 6 above)
Comment (jonniebigodes) [reply content/intro-to-storybook/react/en/simple-component.md:146]: Added with the latest commit. Didn't want to go into it too much, gave a description of what are they and how they will be used. Also left a link pointing to the official documentation, should the reader wants to learn more about it.

Let me know if you're ok with it.

File: content/intro-to-storybook/react/en/conclusion.md [hunk 8]
```diff
@@ -7,19 +7,19 @@ Congratulations! You created your first UI in Storybook. Along the way you learn
...
 
 Want to dive deeper? Here are helpful resources.
 
-- [**Official Storybook documentation**](https://storybook.js.org/basics/introduction/) has API documentation, community links, and the addon gallery.
+- [**Official Storybook documentation**](https://storybook.js.org/docs/react/get-started/introduction) has API documentation, community links, and the addon gallery.
 
 - [**The Delightful Storybook Workflow**](https://blog.hichroma.com/the-delightful-storybook-workflow-b322b76fd07) highlights workflow best practices used by high-velocity teams at Squarespace, Major League Soccer, Discovery Network, and Apollo GraphQL.
 
-- [**Visual Testing Handbook**](https://www.chromatic.com/book/visual-testing-handbook) dives deep into using Storybook to visual test components. Free 31-page ebook.
+- [**Visual Testing Handbook**](https://www.learnstorybook.com/visual-testing-handbook/) dives deep into using Storybook to visual test components. Free 31-page ebook.
```
[the remark this answers is not in the document]
Comment (jonniebigodes) [reply content/intro-to-storybook/react/en/conclusion.md:22]: It's being taken care of in #356. Is not yet merged as some feedback was not addressed.

**Comment by jonniebigodes** [comment]
Merging it now

**Pull request merged**